File: src/main/java/com/gmail/borlandlp/minigamesdtools/gui/other/radar2d/BlockMarker.java
package com.gmail.borlandlp.minigamesdtools.gui.other.radar2d;

import org.bukkit.ChatColor;
import org.bukkit.Location;
import org.bukkit.block.Block;
import org.bukkit.boss.BarColor;

public class BlockMarker implements Marker {
    private ChatColor chatColor;
    private Block block;

    public BlockMarker(Block block, ChatColor color) {
        this.chatColor = color;
        this.block = block;
    }

    @Override
    public boolean isOwner(Object object) {
        return object instanceof Block && ((Block) object).getLocation().toString().equals(this.getLocation().toString());
    }

    @Override
    public ChatColor getColor() {
        return this.chatColor;
    }

    @Override
    public Location getLocation() {
        return this.block.getLocation();
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/gui/other/radar2d/BlockMarker.kt
package com.gmail.borlandlp.minigamesdtools.gui.other.radar2d

import org.bukkit.ChatColor
import org.bukkit.Location
import org.bukkit.block.Block

class BlockMarker(private val block: Block, private val chatColor: ChatColor,
                  override val color: ChatColor,
                  override val location: Location
) :
    Marker {
    override fun isOwner(`object`: Any): Boolean {
        return `object` is Block && `object`.location.toString() == this.location.toString()
    }
}